sdk/elixir/runtime/main.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/iancoleman/strcase"
+)
+
+const (
+	ModSourceDirPath     = "/src"
+	sdkSrc               = "/sdk"
+	genDir               = "sdk"
+	schemaPath           = "/schema.json"
+	defaultElixirVersion = "1.16.1-erlang-26.2.2-debian-bookworm-20240130-slim"
+)
+
+func New(
+	// +optional
+	sdkSourceDir *Directory,
+) *ElixirSdk {
+	return &ElixirSdk{SDKSourceDir: sdkSourceDir, RequiredPaths: []string{}}
+}
+
+type ElixirSdk struct {
+	SDKSourceDir  *Directory
+	RequiredPaths []string
+}
+
+func (m *ElixirSdk) ModuleRuntime(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, err
+	}
+
+	modName, err := modSource.ModuleName(ctx)
+	if err != nil {
+		return nil, err
+	}
+	subPath, err := modSource.SourceSubpath(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	mod := normalizeModName(modName)
+	entrypoint := path.Join(ModSourceDirPath, subPath, mod)
+
+	return ctr.
+		WithEntrypoint([]string{
+			"mix", "cmd",
+			"--cd", entrypoint,
+			"mix do deps.get + dagger.invoke",
+		}), nil
+}
+
+func (m *ElixirSdk) Codegen(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*GeneratedCode, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, fmt.Errorf("could not load module config: %v", err)
+	}
+
+	return dag.GeneratedCode(ctr.Directory(ModSourceDirPath)).
+		WithVCSGeneratedPaths([]string{genDir + "/**"}).
+		WithVCSIgnoredPaths([]string{"dagger"}), nil
```

Comment:
Shouldn't this be `genDir` instead?

Reply:
I just ignore the `dagger` sdk path here. What it's incorrect might be `WithVCSGeneratedPaths`. But if we call `WithVCSIgnoredPaths` with this directory, we don't need to call `WithVCSGeneratedPaths`?

Reply:
Both of them should point to any generated files, including the SDK.

Reply:
Just set to genDir, which's `dagger_sdk`. (fixed in https://github.com/dagger/dagger/pull/6967/commits/b56617b7f36897fb773bdec543f69a645e677be1)